### DISCUSSION TITLE
Backing up AWS RDS postgresql databases

### DISCUSSION THREAD

**Discussion by garybrowne** [prodrigestivill/docker-postgres-backup-local#153] [Q&A] [answered]
Hello,
It's not clear to me if this can be used to backup an AWS RDS postgres database? The "links" and "depends_on" vars in docker compose lead me to believe this is just for containerised local dbs, but hopefully I am mistaken (as the use of psql seems to indicate that this utility would back up any postgres dbs on any host).
Can someone please clarify for me?
Thanks very much,
Gary

**Comment by prodrigestivill** [comment, marked as answer]
Yes, you can use this image, no need to use `links` nor `depends_on`.

Just configure the environment variables to point to your running PostgreSQL (running in RDS or elsewhere) and ensure the routing and firewalls (AWS security groups) allow the connections from this container.

Mainly you would need to setup the following in case you opt to use ECS:
- Environment variables:
  - POSTGRES_HOST: postgres RDS domain name
  - POSTGRES_DB
  - POSTGRES_USER
  - POSTGRES_PASSWORD: ideally using secrets
- Data Volumes: ([ECS Storage Options](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_data_volumes.html))
  -  A **persistent volume** (EBS, EFS, ...) to store the backups at `/backups`.

**Reply by prodrigestivill** [reply]
Also keep in mind you might also be interested in using the [schickling/postgres-backup-s3](https://hub.docker.com/r/schickling/postgres-backup-s3) image.
